### DISCUSSION TITLE
Specify custom theme types

### DISCUSSION THREAD

**Discussion by goncy** [chakra-ui/chakra-ui#2441] [Q&A] [answered]
How can I specify which colors my theme has so I can get autocomplete for them on the rest of the app using typescript?

**Comment by mdovn** [comment]
I believe that's not possible right now since we can change the theme (context) at runtime. Which make it hard to type checking.

**Reply by goncy** [reply]
I was thinking something like a .d.ts file specifying the theme type

**Reply by mdovn** [reply]
I actually thought about that idea but never try yet.

**Comment by rockymeza** [comment]
This might be interesting for when Chakra UI upgrades to Emotion 11

https://emotion.sh/docs/emotion-11#theme-type

**Comment by with-heart** [comment, marked as answer]
This is currently not possible. There's no way for our component types to read anything custom that you've added to your theme (like components, variants, colors, etc.). We would need to build tooling for local code generation to enable that, which I've thought a lot about over the last few months.

That's a serious level of effort that we need to plan for. I'd love to be able to offer that in the future though, and hope to explore the idea more after v1 is released.

**Reply by goncy** [reply]
Great, thanks, just wanted to know if it was available but I wasn't able to find it

**Reply by d-strygwyr** [reply]
any update for this feature 😁 ?
